Lesson 6 - Developer tools/code/server.js

```diff
@@ -16,11 +16,53 @@ app.get('/today', function (req, res) {
 });
 
 //POST /numbers should save number for later
+app.post('/numbers', function (req, res) {
+  numbers.push(+req.body);
+  res.send('Number stored');
+});
+
 //GET /numbers should return comma separated list of numbers
+app.get('/numbers', function (req, res) {
+    setTimeout(function () {
+        res.send(numbers.join(", "));
+    }, 1000);
+});
+
 //DELETE /numbers should empty numbers array
+app.delete('/numbers', function (req, res) {
+  numbers = [];
+  res.send('Numbers deleted');
+});
+
 //GET /numbers/:op should return result of given operation of stored numbers (check request.params)
 // available operations are sum, difference, multiplication, division
 // return result as strings
+app.get('/numbers/:op', function (req, res) {
+    setTimeout(function () {
+        if (numbers.length === 0) {
+            res.send('0');
+            return;
+        }
+        
+        switch(req.params.op) {
+            case 'sum' : 
+                res.send(numbers.reduce((a, b) => a + b, 0).toString());
```

Comment:
Got some problems on my `node` version with lambdas here. You could put some `"engine": "node >= ^4.1.x"` info in `package.json`